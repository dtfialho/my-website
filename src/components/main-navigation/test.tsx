import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Nav from './'

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '',
      pathname: '',
      query: '',
      asPath: ''
    }
  }
}))

const useRouter = jest.spyOn(require('next/router'), 'useRouter')

describe('MainNavigation', () => {
  it('Should render correctly', () => {
    useRouter.mockImplementation(() => ({
      asPath: '/'
    }))

    const { container } = render(<Nav />)

    expect(container).toMatchSnapshot()
  })

  it('Should set active link correctly', () => {
    useRouter.mockImplementation(() => ({
      asPath: '/about-me'
    }))

    render(<Nav />)

    userEvent.click(screen.getByRole('button', { name: /Menu/ }))

    expect(screen.getByRole('link', { name: /Sobre Mim/ })).toHaveStyleRule(
      'width',
      '100%',
      {
        modifier: '::after'
      }
    )
  })
})
