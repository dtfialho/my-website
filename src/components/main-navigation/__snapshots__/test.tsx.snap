// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNavigation Should render correctly 1`] = `
.c6 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
  background: none;
  cursor: pointer;
}

.c0 {
  background: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
  width: 50px;
  height: 50px;
  padding: 5px;
  position: fixed;
  top: 20px;
  right: 40px;
  border-radius: 50%;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  position: relative;
  display: block;
  background: #fff;
  width: 40px;
  height: 3px;
  -webkit-transition: 0.5s ease-in-out;
  transition: 0.5s ease-in-out;
}

.c1::before,
.c1::after {
  background: #fff;
  content: '';
  display: block;
  width: 100%;
  height: 100%;
  position: absolute;
  -webkit-transition: 0.5s ease-in-out;
  transition: 0.5s ease-in-out;
}

.c1::before {
  top: -12px;
}

.c1::after {
  bottom: -12px;
}

.c2 {
  visibility: hidden;
  opacity: 0;
  list-style: none;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin: 0 auto;
  height: 0;
}

.c3 {
  display: block;
  text-align: center;
}

.c4 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #fff;
  font-weight: 700;
  font-size: 20px;
  position: relative;
}

.c4:hover::after {
  width: 100%;
}

.c4::after {
  display: block;
  width: 100%;
  height: 2px;
  -webkit-transition: width 0.3s ease-in-out;
  transition: width 0.3s ease-in-out;
  background-color: #fff;
  content: '';
}

.c5 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #fff;
  font-weight: 700;
  font-size: 20px;
  position: relative;
}

.c5:hover::after {
  width: 100%;
}

.c5::after {
  display: block;
  width: 0;
  height: 2px;
  -webkit-transition: width 0.3s ease-in-out;
  transition: width 0.3s ease-in-out;
  background-color: #fff;
  content: '';
}

@media (min-width:768px) {
  .c0 {
    display: none;
  }
}

@media (min-width:768px) {
  .c2 {
    max-width: 1200px;
    width: auto;
    height: auto;
    display: block;
    visibility: visible;
    opacity: 1;
  }
}

@media (min-width:768px) {
  .c3 {
    display: inline-block;
    padding: 0 20px;
  }
}

@media (max-width:768px) {
  .c3 {
    margin-bottom: 5px;
  }
}

@media (max-width:768px) {
  .c4::after {
    margin-top: 5px;
  }
}

@media (max-width:768px) {
  .c5::after {
    margin-top: 5px;
  }
}

<div>
  <nav
    class=""
  >
    <button
      class="c0"
      title="Menu"
      type="button"
    >
      <span
        class="c1"
      />
    </button>
    <ul
      class="c2"
    >
      <li
        class="c3"
      >
        <a
          class="c4"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="c3"
      >
        <a
          class="c5"
          href="/blog"
        >
          Blog
        </a>
      </li>
      <li
        class="c3"
      >
        <a
          class="c5"
          href="/about-me"
        >
          Sobre mim
        </a>
      </li>
      <li
        class="c3"
      >
        <button
          class="c6"
          type="button"
        >
          <img
            alt="Active language"
            height="22"
            src="/img/pt-BR.jpg"
            width="29"
          />
        </button>
      </li>
    </ul>
  </nav>
</div>
`;
